lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -146,15 +172,17 @@ private NettyMessagingTransport(
     this.clientBootstrap.group(this.clientWorkerGroup)
         .channel(NioSocketChannel.class)
         .handler(new NettyChannelInitializer(new NettyDefaultChannelHandlerFactory("client",
-            this.clientChannelGroup, this.clientEventListener)))
+            this.clientChannelGroup, this.clientEventListener), sslContextClient,
+                protocolType.equals(PROTOCOL_TCP) ? ChannelType.TCP : ChannelType.HTTP_CLIENT))
```

Comment:
Implement a method in `ChannelType` to get the channel type by protocol, so you can call e.g. `ChannelType.Client(protocolType)`